app/models/statement_creator.rb

```diff
@@ -0,0 +1,63 @@
+class StatementCreator
+  attr_accessor :order_detail_ids, :errors, :to_statement, :account_statements, :session_user, :current_facility
+
+  def initialize(params)
+    @order_detail_ids = params[:order_detail_ids]
+    @session_user = params[:session_user]
+    @current_facility = params[:current_facility]
+    @errors = []
+    @to_statement = {}
+  end
+
+  def create
+    OrderDetail.transaction do
+      set_order_details_to_statement
+      setup_statement_from_details
+    end
+  end
+
+  def formatted_errors
+    errors.join("<br/>")
```

Comment:
I could go either way, but part of me feels like the `join` belongs more in the controller than the service object, especially since we have two of these, and they're HTML specific. I don't have a strong feeling, though.